crates/ty_python_semantic/src/semantic_index/use_def/place_state.rs

```diff
@@ -207,6 +230,15 @@ impl Bindings {
         self.unbound_narrowing_constraint
             .unwrap_or(self.live_bindings[0].narrowing_constraint)
     }
+
+    pub(super) fn mark_reachability_constraints(
+        &self,
+        reachability_constraints: &mut ReachabilityConstraintsBuilder,
+    ) {
+        for binding in &self.live_bindings {
+            reachability_constraints.mark_used(binding.reachability_constraint);
+        }
```

Comment:
Same here. I think it would be ncie if we could call `shrink_to_fit` on `live_bindings`.  This seems easy now given that we mark them as part of an extra traversal. 

See https://github.com/astral-sh/ty/issues/850